storybookjs/storybook
# Autodocs: Fix docs pages ignoring `meta.id` when calculating their ID

Closes #23394

## What I did

I made a little adjustment in `StoryIndexGenerator` so that the docs pages that are automatically generated would also prioritize `Meta.id` over `Meta.title` if `Meta.id` was provided.

I referenced code section below for the adjustment.

https://github.com/storybookjs/storybook/blob/c8e9caf9514cdb021834c0bbe6082faca733381a/code/lib/csf-tools/src/CsfFile.ts#L469-L493

## How to test

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Add `id` to `meta` object in `Button.stories.ts` (e.g., `MyButton`)
3. Open Storybook in your browser
4. Go to `Button` docs page and confirm the url is something like: `http://localhost:6007/?path=/docs/mybutton--docs`

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers


`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`
